libgstd/gstd_parser.c

```diff
@@ -966,3 +980,209 @@ gstd_parser_signal_timeout (GstdSession * session, gchar * action, gchar * args,
 
   return ret;
 }
+
+static GstdReturnCode
+gstd_parser_pipeline_create_ref (GstdSession * session, gchar * action,
+    gchar * args, gchar ** response)
+{
+  gchar **tokens = NULL;
+  gchar *description = NULL;
+  GstdObject *pipeline_list_node = NULL;
+  GstdObject *pipeline_node = NULL;
+  GstdReturnCode ret = GSTD_EOK;
+
+  g_return_val_if_fail (GSTD_IS_SESSION (session), GSTD_NULL_ARGUMENT);
+  g_return_val_if_fail (action, GSTD_NULL_ARGUMENT);
+  g_return_val_if_fail (args, GSTD_NULL_ARGUMENT);
+  g_return_val_if_fail (response, GSTD_NULL_ARGUMENT);
+
+  tokens = g_strsplit (args, " ", 2);
+
+  /* Get the pipeline list node */
+  ret = gstd_get_by_uri (session, "/pipelines", &pipeline_list_node);
+  if (ret) {
+    goto pipeline_list_node_error;
+  }
```

Comment:
why the command to get the pipeline list is not inside the session  lock-unlock ?

Reply:
I try to keep the locked sections to a minimal to avoid stalls and couldn't think of a case where we needed to protect the pipeline list.
Is there a way that the pipeline list pointer can change between parser calls?